File: pdftract-dotnet/src/Pdftract/Models/Classification.cs
using MessagePack;

namespace Pdftract.Models;

/// <summary>
/// Represents document classification results.
/// </summary>
[MessagePackObject]
public record Classification
{
    /// <summary>
    /// Document category classification.
    /// </summary>
    [Key(0)]
    public required string Category { get; init; }

    /// <summary>
    /// Confidence score for the classification (0.0 to 1.0).
    /// </summary>
    [Key(1)]
    public required double ConfidenceScore { get; init; }
}

File: sdk/php/src/Pdftract/Models/Classification.php
<?php

declare(strict_types=1);

namespace Jedarden\Pdftract\Models;

/**
 * Readonly classification model
 *
 * Simple readonly representation of document classification results
 */
class Classification
{
    /**
     * @param string $type Classification type (e.g., "invoice", "contract", "report")
     * @param float $confidence Confidence score between 0.0 and 1.0
     */
    public function __construct(
        public readonly string $type,
        public readonly float $confidence
    ) {}
}
